exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionSenderRootExec.java

```diff
@@ -358,7 +357,7 @@ public void close() throws Exception {
     }
   }
 
-  public void sendEmptyBatch(boolean isLast) {
+  public synchronized void sendEmptyBatch(boolean isLast) {
```

Comment:
Method can be made private. Plus maybe not be synchronized as well since is being called from `innerNext`.

Reply:
Done